business_objects/monitor.py

```diff
@@ -0,0 +1,424 @@
+    ) tasks
+    JOIN project p
+    ON p.id = tasks.project_id
+    JOIN organization orga
+    ON orga.id = p.organization_id
+    """
+
+    return general.execute_all(query)
+
+
+def cancel_all_running_tasks(project_id: str = None):
+    set_information_source_payloads_to_failed(project_id)
+    set_attribute_calculation_to_failed(project_id)
+    set_embedding_to_failed(project_id)
+    set_record_tokenization_task_to_failed(project_id)
+    set_upload_task_to_failed(project_id)
+    set_weak_supervision_to_failed(project_id)
+    general.commit()
+
+
+def set_information_source_payloads_to_failed(
+    project_id: str = None, payload_id: str = None, with_commit: bool = False
+) -> None:
+    query = f"""
+    UPDATE information_source_payload
+    SET state = '{enums.PayloadState.FAILED.value}'
+    WHERE state = '{enums.PayloadState.STARTED.value}'
+    """
+
+    if project_id:
+        query = (
+            query
+            + f"""
+        AND project_id = '{project_id}'
+        """
+        )
+
+    if payload_id:
+        query = (
+            query
+            + f"""
+        AND id = '{payload_id}'
+        """
+        )
+    general.execute(query)
+    general.flush_or_commit(with_commit)
+
+
+def set_attribute_calculation_to_failed(
+    project_id: str = None, attribute_id: str = None, with_commit: bool = False
+) -> None:
+    query = f"""
+    UPDATE attribute
+    SET state = '{enums.AttributeState.FAILED.value}'
+    WHERE state = '{enums.AttributeState.RUNNING.value}'
+    """
+
+    if project_id:
+        query = (
+            query
+            + f"""
+        AND project_id = '{project_id}'
+        """
+        )
+    if attribute_id:
+        query = (
+            query
+            + f"""
+        AND id = '{attribute_id}'
+        """
+        )
+    general.execute(query)
+    general.flush_or_commit(with_commit)
+
+
+def set_record_tokenization_task_to_failed(
+    project_id: str = None, task_id: str = None, with_commit: bool = False
+) -> None:
+    query = f"""
+    UPDATE record_tokenization_task
+    SET state = '{enums.TokenizerTask.STATE_FAILED.value}'
+    WHERE state = '{enums.TokenizerTask.STATE_IN_PROGRESS.value}'
+    """
+
+    if project_id:
+        query = (
+            query
+            + f"""
+        AND project_id = '{project_id}'
+        """
+        )
+    if task_id:
+        query = (
+            query
+            + f"""
+        AND id = '{task_id}'
+        """
+        )
+    print(query)
+    general.execute(query)
+    general.flush_or_commit(with_commit)
+
+
+def set_embedding_to_failed(
+    project_id: str = None, embedding_id: str = None, with_commit: bool = False
+) -> None:
+    query = f"""
+    UPDATE embedding
+    SET state = '{enums.EmbeddingState.FAILED.value}'
+    WHERE state = '{enums.EmbeddingState.ENCODING.value}' OR state = '{enums.EmbeddingState.WAITING.value}' OR state = '{enums.EmbeddingState.INITIALIZING.value}'
```

Comment:
this doesn't work as expected since OR has a lower precedence than AND. We either need a brackets or (preferred) since this is on one field the IN operator

Reply:
Done